### ISSUE TITLE
Expo Fabric Reanimated Crash on iOS - Metro Missing Global `fabric: true` Variable

### ISSUE THREAD

**Issue by alexhernandez** [expo/expo#23994]
### Minimal reproducible example

https://github.com/alexhernandez/expo-fabric-crash-example

### Summary

A fresh react native repo with `expo bare modules` and `react-native-reanimated` causes a crash (on iOS only) with the error message:
```
 ERROR  TypeError: Cannot read property 'stateNode' of undefined

This error is located at:
    in AnimatedComponent(View)
    in Unknown (created by App)
    in RCTScrollContentView (created by ScrollView)
    in RCTScrollView (created by ScrollView)
    in ScrollView (created by ScrollView)
    in ScrollView (created by App)
    in RCTSafeAreaView (created by App)
    in App
    in RCTView (created by View)
    in View (created by AppContainer)
    in RCTView (created by View)
    in View (created by AppContainer)
    in AppContainer
    in ExampleProject(RootComponent), js engine: hermes
```
iOS:
![Screenshot 2023-08-16 at 5 58 20 PM](https://github.com/expo/expo/assets/2424355/e9e91c7e-0c24-4384-bd1f-b503473a2bcf)

Android:
![Screenshot 2023-08-16 at 5 14 54 PM](https://github.com/expo/expo/assets/2424355/4a4b5c3c-ed4c-45c5-b93e-b4fe6947a153)

---

Crash is possibly due to a missing global variable `fabric: true`, which Is present in the [react native docs](https://reactnative.dev/docs/new-architecture-app-intro#running-the-app), but not inside of Metro Bundler logs:
```
 LOG  Running "ExampleProject" with {"rootTag":1,"initialProps":{"concurrentRoot":true}}
```

This issue does NOT occur on iOS or Android when `expo bare modules` is not present.

iOS:
![Screenshot 2023-08-16 at 5 10 47 PM](https://github.com/expo/expo/assets/2424355/ae9fa996-372a-430e-9a22-8581159325fd)

Android:
![Screenshot 2023-08-16 at 5 14 54 PM](https://github.com/expo/expo/assets/2424355/4a4b5c3c-ed4c-45c5-b93e-b4fe6947a153)

### Environment

npx expo-env-info:

```
    System:
      OS: macOS 13.5
      Shell: 5.9 - /bin/zsh
    Binaries:
      Node: 18.16.1 - ~/.nvm/versions/node/v18.16.1/bin/node
      Yarn: 1.22.19 - ~/.yarn/bin/yarn
      npm: 9.5.1 - ~/.nvm/versions/node/v18.16.1/bin/npm
      Watchman: 2023.08.07.00 - /usr/local/bin/watchman
    Managers:
      CocoaPods: 1.12.1 - /Users/xxx/.rbenv/shims/pod
    SDKs:
      iOS SDK:
        Platforms: DriverKit 22.4, iOS 16.4, macOS 13.3, tvOS 16.4, watchOS 9.4
    IDEs:
      Android Studio: 2022.3 AI-223.8836.35.2231.10406996
      Xcode: 14.3.1/14E300c - /usr/bin/xcodebuild
    npmPackages:
      expo: 49.0.7 => 49.0.7 
      react: 18.2.0 => 18.2.0 
      react-native: 0.72.4 => 0.72.4 
    Expo Workflow: bare
    ```
    
npx react-native info:

    ```
    System:
  OS: macOS 13.5
  CPU: (16) x64 Intel(R) Core(TM) i9-9980HK CPU @ 2.40GHz
  Memory: 9.67 GB / 64.00 GB
  Shell:
    version: "5.9"
    path: /bin/zsh
Binaries:
  Node:
    version: 18.16.1
    path: ~/.nvm/versions/node/v18.16.1/bin/node
  Yarn:
    version: 1.22.19
    path: ~/.yarn/bin/yarn
  npm:
    version: 9.5.1
    path: ~/.nvm/versions/node/v18.16.1/bin/npm
  Watchman:
    version: 2023.08.07.00
    path: /usr/local/bin/watchman
Managers:
  CocoaPods:
    version: 1.12.1
    path: /Users/xxx/.rbenv/shims/pod
SDKs:
  iOS SDK:
    Platforms:
      - DriverKit 22.4
      - iOS 16.4
      - macOS 13.3
      - tvOS 16.4
      - watchOS 9.4
  Android SDK: Not Found
IDEs:
  Android Studio: 2022.3 AI-223.8836.35.2231.10406996
  Xcode:
    version: 14.3.1/14E300c
    path: /usr/bin/xcodebuild
Languages:
  Java:
    version: 11.0.20
    path: /usr/bin/javac
  Ruby:
    version: 3.1.4
    path: /Users/xxx/.rbenv/shims/ruby
npmPackages:
  "@react-native-community/cli": Not Found
  react:
    installed: 18.2.0
    wanted: 18.2.0
  react-native:
    installed: 0.72.4
    wanted: 0.72.4
  react-native-macos: Not Found
npmGlobalPackages:
  "*react-native*": Not Found
Android:
  hermesEnabled: true
  newArchEnabled: true
iOS:
  hermesEnabled: true
  newArchEnabled: true
  ```

**Comment by alexhernandez** [comment]
Issue possible related to issue here: https://github.com/expo/expo/issues/23905

**Comment by alexhernandez** [comment]
Might also be good for Reanimated team to be aware of:
@honcon @j-piasecki

**Comment by steve228uk** [comment]
Also causing an issue with `react-native-screens` https://github.com/software-mansion/react-native-screens/issues/1809#issuecomment-1680649713

**Comment by rikkrome** [comment]
What release version will this fix be in?

**Comment by sharon-yi** [comment]
@Kudo also waiting for the release for this fix!

**Comment by Kudo** [comment]
published in `expo@49.0.8`
